rama-http/src/service/web/router.rs

```diff
@@ -0,0 +1,151 @@
+use std::{collections::HashMap, convert::Infallible, sync::Arc};
+
+use crate::{Method, Request, Response};
+
+use matchit::Router as MatchitRouter;
+use rama_core::{
+    Context,
+    service::{BoxService, Service},
+};
+use rama_http_types::Body;
+
+pub struct Router<State> {
+    routes: HashMap<Method, MatchitRouter<Arc<BoxService<State, Request, Response, Infallible>>>>,
```

Comment:
```suggestion
    routes: MatchitRouter<Vec<(HttpMatcher, BoxService<State, Request, Response, Infallible>)>>,
```

I would suggest to try this approach. Where you first match based on path and when found use the first service whose HttpMatcher matches the input.

Cool thing about the http matcher is that it allows not only a method, but also multiple methods, and other more complex matching cases.

Reply:
interesting, I will take a look at this!